mne/viz/backends/_ipyvolume.py

```diff
@@ -0,0 +1,372 @@
+            plot(x, y, z, color=color)
+
+    def surface(self, surface, color=None, opacity=1.0,
+                vmin=None, vmax=None, colormap=None, scalars=None,
+                backface_culling=False):
+        from matplotlib import cm
+        from matplotlib.colors import ListedColormap
+
+        if colormap is None:
+            cmap = cm.get_cmap('coolwarm')
+        else:
+            cmap = ListedColormap(colormap / 255.0)
+
+        vertices = np.array(surface['rr'])
+        x, y, z = vertices.T
+        triangles = np.array(surface['tris'])
+
+        if scalars is not None:
+            if vmin is None:
+                vmin = min(scalars)
+            if vmax is None:
+                vmax = max(scalars)
+            nscalars = (scalars - vmin) / (vmax - vmin)
+            color = cmap(nscalars)
+        else:
+            color = np.append(color, opacity)
+
+        mesh = ipv.plot_trisurf(x, y, z, triangles=triangles, color=color)
+        _add_transperent_material(mesh)
+
+    def sphere(self, center, color, scale, opacity=1.0, resolution=8,
+               backface_culling=False):
+        default_sphere_radius = 0.5
+        if center.ndim == 1:
+            center = np.array([center])
+
+        # fetch original data
+        orig_vertices, orig_faces = \
+            _create_sphere(radius=scale * default_sphere_radius,
+                           cols=resolution, rows=resolution)
+        n_vertices = len(orig_vertices)
+        orig_vertices = np.c_[orig_vertices, np.ones(n_vertices)].T
+        n_faces = len(orig_faces)
+        n_center = len(center)
+        voffset = 0
+        foffset = 0
+
+        # accumulate mesh data
+        acc_vertices = np.empty((n_center * n_vertices, 3), dtype=np.float32)
+        acc_faces = np.empty((n_center * n_faces, 3), dtype=np.uint32)
+        for c in center:
+            # apply index shifting and accumulate faces
+            current_faces = orig_faces + voffset
+            acc_faces[foffset:foffset + n_faces, :] = current_faces
+            foffset += n_faces
+
+            # apply translation and accumulate vertices
+            mat = np.eye(4)
+            mat[:-1, 3] = c
+            current_vertices = mat.dot(orig_vertices)
+            acc_vertices[voffset:voffset + n_vertices, :] = \
+                current_vertices.T[:, 0:3]
+            voffset += n_vertices
+
+        x, y, z = acc_vertices.T
+        color = np.append(color, opacity)
+
+        ipv.squarelim()
+        mesh = ipv.plot_trisurf(x, y, z, triangles=acc_faces, color=color)
+        _add_transperent_material(mesh)
+
+    def quiver3d(self, x, y, z, u, v, w, color, scale, mode, resolution=8,
+                 glyph_height=None, glyph_center=None, glyph_resolution=None,
+                 opacity=1.0, scale_mode='none', scalars=None,
+                 backface_culling=False):
+        color = np.append(color, opacity)
+        x, y, z = map(_check_array, [x, y, z])
```

Comment:
What are the advantages over `map()`?

Reply:
It is more readable, and in some cases (i.e. custom functions) it can be faster. I don't think that it would be much speed difference in this case. You could benchmark it if you are curious about it.

But readability is enough for me, even if the performance is much lower (unless this line is called many times, but I did not profile it).

Reply:
what do you think @agramfort ?

Reply:
yes it's readability

Reply:
I mean I find map more readable here. I asked for it :)